File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price, max_profit = prices[0], 0

        for price in prices:
            if price < min_price:
                min_price = price
            else :
                max_profit = max(max_profit, price - min_price)
        
        return max_profit

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if (nums.size() == 1) return true;

        int maxReach = 0;
        int n = nums.size();

        for (int i = 0; i < n; ++i) {
            if (i >maxReach) return false;
            maxReach = max(maxReach, i + nums[i]);
            if (maxReach > n - 1) return true;
        }
        return true;
    }
};
